src/sage/groups/finitely_presented.py

```diff
@@ -1532,7 +1638,184 @@ def alexander_matrix(self, im_gens=None):
+          ``self.abelianization_to_algebra()`` to avoid recomputation
+        - ``simplified`` -- boolean (default: ``False``); if set to
+          ``True`` use Gauss elimination and erase rows and columns
+
+        OUTPUT:
+
+        - ``A`` -- a matrix with coefficients in ``R``
+
+        - ``R`` -- a ring of Laurent polynomials over ``ring``
+
+        - ``ideal`` -- an list of generators of an ideal ``I`` of ``R`` such that ``R/I`` is
+          the group algebra of the abelianization of ``self``
+
+        EXAMPLES::
+
+            sage: G.<a,b,c> = FreeGroup()
+            sage: H = G.quotient([a*b/a/b, a*c/a/c, c*b/c/b])
+            sage: A, R, ideal = H.abelian_alexander_matrix()
+            sage: A
+            [-f2 + 1  f1 - 1       0]
+            [-f3 + 1       0  f1 - 1]
+            [      0  f3 - 1 -f2 + 1]
+            sage: R
+            Multivariate Laurent Polynomial Ring in f1, f2, f3 over Rational Field
+            sage: ideal
+            []
+            sage: abel = H.abelianization_to_algebra()
+            sage: (A, R, ideal) == H.abelian_alexander_matrix(abelianized=abel)
+            True
+            sage: G=FreeGroup(3)/[(2,1,1), (1,2,2,3,3)]
+            sage: A, R, ideal = G.abelian_alexander_matrix(simplified=True); A
+            [-f3^2 - f3^4 - f3^6         f3^3 + f3^6]
+            sage: g=FreeGroup(1)/[]
+            sage: g.abelian_alexander_matrix()
+            ([], Integer Ring, [])
+            sage: g=FreeGroup(0)/[]
+            sage: g.abelian_alexander_matrix()
+            ([], Integer Ring, [])
+        """
+        if abelianized is None:
+            ab, R, ideal, images = self.abelianization_to_algebra(ring=ring)
+        else:
+            ab, R, ideal, images = abelianized
+        A = self.alexander_matrix(im_gens=images)
+        if simplified:
+            n, m = A.dimensions()
+            R = A.base_ring()
+            if 0 in (n, m):
+                return A, R, ideal
+            simpli = True
+            while simpli:
+                i = 0
+                j = 0
+                unidad = False
+                while not unidad and i < n and j < m:
+                    p = A[i, j]
+                    unidad = p.is_unit()
+                    if unidad:
+                        A.swap_rows(0, i)
+                        A.swap_columns(0, j)
+                        for k in range(1, n):
+                            A.add_multiple_of_row(k, 0, -A[k, 0] * p ** -1)
+                        A = A.delete_rows([0]).delete_columns([0])
+                        n, m = A.dimensions()
+                    else:
+                        if j < m - 1:
+                            j += 1
+                        else:
+                            i += 1
+                            j = 0
+                simpli = unidad
+        return A, R, ideal
+
+    def char_var(self, ring=QQ, matrix_ideal=None, abelianized=None, groebner=False):
```

Comment:
```suggestion
    def characteristic_varieties(self, ring=QQ, matrix_ideal=None, abelianized=None, groebner=False):
```
This needs to be spelled out.

Reply:
I changed it here and in the other previous occurances of  `char_ var`